File: app/src/main/java/com/dylanc/viewbinding/sample/base/java/BaseBindingAdapter.java
package com.dylanc.viewbinding.sample.base.java;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewbinding.ViewBinding;

import com.dylanc.viewbinding.BindingViewHolder;

import java.util.ArrayList;
import java.util.List;

public abstract class BaseBindingAdapter<VB extends ViewBinding> extends RecyclerView.Adapter<BindingViewHolder<VB>> {

  @NonNull
  @Override
  public BindingViewHolder<VB> onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    return BindingViewHolder.createWithGeneric(this, parent);
  }
}

File: app/src/main/java/com/dylanc/viewbinding/sample/base/kotlin/BaseBindingAdapter.kt
package com.dylanc.viewbinding.sample.base.kotlin

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import androidx.viewbinding.ViewBinding
import com.dylanc.viewbinding.BindingViewHolder
import com.dylanc.viewbinding.newBindingViewHolderWithGeneric

abstract class BaseBindingAdapter<VB : ViewBinding, T> : RecyclerView.Adapter<BindingViewHolder<VB>>() {

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = newBindingViewHolderWithGeneric<VB>(parent)
}